service/worker/deployment/deployment_workflow.go

```diff
@@ -97,14 +135,33 @@ func (d *DeploymentWorkflowRunner) run() error {
 
 	selector := workflow.NewSelector(d.ctx)
 	selector.AddReceive(updateDeploymentSignalChannel, func(c workflow.ReceiveChannel, more bool) {
-		// Process Signal
-		forceCAN = true
+		// fetch the input from the signal
+		var signalInput *deployspb.UpdateDeploymentSignalInput
+		updateDeploymentSignalChannel.Receive(d.ctx, &signalInput)
+
+		if d.DeploymentLocalState.TaskQueueFamilies == nil {
```

Comment:
I think you need separate checks for `d.DeploymentLocalState.TaskQueueFamilies == nil` and `d.DeploymentLocalState.TaskQueueFamilies[signalInput.Name] == nil`, right?

Reply:
Won't `d.DeploymentLocalState.TaskQueueFamilies[signalInput.Name] == nil` will always be true whenever we have `d.DeploymentLocalState.TaskQueueFamilies == nil` no?

This line of code will anyways be hit only when we are going to add a key in our map, so I thought that check was repeated (unless I've missed something)

Reply:
Yes, but it can also be true if the first is not true

Reply:
ah, that's stupid from me, sorry about this. Changed.